libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2333,3 +2333,34 @@ string YulUtilFunctions::extractReturndataFunction()
 	});
 }
 
+string YulUtilFunctions::copyConstructorArgumentsToMemoryFunction(
+	ContractDefinition const& _contract,
+	string const& _creationObjectName
+)
+{
+	string functionName = "copy_constructor_arguments_" + _creationObjectName;
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		string returnParams = suffixedVariableNameList("ret_param_",0, _contract.constructor()->parameters().size());
+		ABIFunctions abiFunctions(m_evmVersion, m_revertStrings, m_functionCollector);
+
+		return util::Whiskers(R"(
+			function <functionName>() -> <retParams> {
+				let programSize := datasize("<object>")
+				let argSize := sub(codesize(), programSize)
+
+				let memoryDataOffset := <allocate>(argSize)
+				codecopy(memoryDataOffset, programSize, argSize)
+
+				<retParams> := <abiDecode>(memoryDataOffset, add(memoryDataOffset, argSize))
+			}
+		)")
+		("functionName", functionName)
+		("retParams", returnParams)
+		("object", _creationObjectName)
+		("allocate", allocationFunction())
+		("abiDecode", abiFunctions.tupleDecoder(_contract.constructor()->functionType(false)->parameterTypes(), true))
```

Comment:
I'm not sure if this makes a difference, but the old code generator uses `FunctionType(_contract.constructor())` - where the main difference is that that one uses Kind::Declaration while this one is Kind::External.

Reply:
Ah, no, it does not make a difference.

Reply:
I am not sure if it makse difference, but it is more clean like you suggested, so will fix